### PR TITLE
[JSON] add ActivateScreenSaver

### PR DESCRIPTION
**Pull request by DaVukovic**
## Description

This changes introduces the option to activate the currently used screensaver. 
It's already available as a Python built-in and I thought it might be a good idea to have that as well. 

If this is not the correct way to implement that, then please let me know how to do it correct. For example, I'm not so sure if the version number I`m using is correct of if it should be 12.5.0
So please let me know if anything is wrong or could be done better.

## Motivation and context

solves: #20088

## How has this been tested?

I did a few tests with Kodi on Linux and also tried with an additionally installed screensaver add-on (bing fotos of the week). 

## What is the effect on users?

At least nothing negative ;). Users are simply able to activate the screensaver via a JSON call. It doesn't break anythin. 

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [x] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [ ] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [x] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

**Comment by Tolriq** [comment]
Should be 12.5.0 as API change :) Beware that as soon as there's JSON divergence with Matrix future JSON patch won't be able to be backported so maybe it should be backported too as simple to keep backports opened.

About "namespace" is this really System and not more GUI? I suppose this would be a no-op on a future GUI less Kodi. (The built in does not have System.ActivateScreenSaver it's just labelled as that in the Wiki.

BTW if you are in adding the missing stuff here's the biggest one that can't be used from JSON and still require Event Server to run the built in.
```
PlayerControl(tempoup)
PlayerControl(tempodown)

CECActivateSource
CECStandby
CECToggleState

PlayDisc
RipCD
```

**Comment by garbear** [comment]
BTW, the commit without whitespace changes is easier to review, here: https://github.com/xbmc/xbmc/commit/9b9db66af648e79dd807ca515a00b31a7105764b?w=1 (appended `?w=1` to raw commit)

**Comment by DaVukovic** [comment]
@Tolriq 
Thanks for your input

>About "namespace" is this really System and not more GUI? I suppose this would be a no-op on a future GUI less Kodi. (The built in does not have System.ActivateScreenSaver it's just labelled as that in the Wiki.

True. My bad. I thought one needs to trigger it via `System.ActivateScreenSaver`. But `ActivateScreenSaver` is what one needs to call. So I will change that to GUI-Operations. 

Will do a backport of that for sure if it gets accepted. 

@garbear 
>BTW, the commit without whitespace changes is easier to review, ....

That was a mistake by me. I force-pushed without commiting ;). Hence the commit. Please ignore that. I'll do an update of this PR anyway.

**Comment by garbear** [comment]
> That was a mistake by me. I force-pushed without commiting ;). Hence the commit. Please ignore that. I'll do an update of this PR anyway.

If you are a git-fu master, you'll do a "cleanup" commit first and a "meaty" commit after, so that the meaty commit has exactly what needs to be reviewed ;)

**Comment by DaVukovic** [comment]
> If you are a git-fu master

Well, I'm not ;)

I've just pushed the latest changes. If this PR is hard to review, I'll happily close it and create a new one.

**Review by garbear — APPROVED** [review]
[Link to new commit](https://github.com/xbmc/xbmc/pull/20110/commits/e05f9f13f6e4c6b0d8cd676aea1a6e9a77607998). I'm not sure about the version number, if you should bump the minor or the patch. But otherwise this looks like a clean addition of a feature that makes sense.

**Review by Montellese — COMMENTED** [review]
Looks good apart from the unnecessary whitespace changes in `JSONServiceDescription.cpp`.

**Comment by DaVukovic** [comment]
Yeah, that's what Jenkins suggests

**Comment by lrusak** [comment]
@DaVukovic would it be possible for you to split out the clang-format changes for `CJSONServiceDescription::m_methodMaps` into a pre commit? This way your commit changes exactly what is needed.

**Review by DaveTBlake — CHANGES_REQUESTED** [review]
Unhappy with having whitespace changes throughout JSONServiceDescription.cpp in this PR. 
Since it is part of our coding standard to allow additional whitespace in code for readability of things like data structures, I would suggest that the necessary `// clang-format on` `// clang-format off` are added for this instead of stripping whitespace. Otherwise the addition of the method is fine.  

Note merging this will mark the end of fixes to JSON API in Matrix. There are several other API changes waiting to merge that have been delayed for this reason. Perhaps we should merge them in order of creation rather than give this priority? The only reason to do this one first would be if we also backport it to v19? Any opinions?

File: xbmc/interfaces/json-rpc/JSONServiceDescription.cpp [hunk 1]
```diff
@@ -38,6 +38,7 @@ std::map<std::string, JSONSchemaTypeDefinitionPtr> CJSONServiceDescription::m_ty
 CJSONServiceDescription::IncompleteSchemaDefinitionMap CJSONServiceDescription::m_incompleteDefinitions = CJSONServiceDescription::IncompleteSchemaDefinitionMap();
 
 JsonRpcMethodMap CJSONServiceDescription::m_methodMaps[] = {
+// clang-format off
```
Comment (howie-f) [inline xbmc/interfaces/json-rpc/JSONServiceDescription.cpp:41]: only need to move this line above `JsonRpcMethodMap…` and we‘re done. thanks

**Comment by DaVukovic** [comment]
@howie-f 

Obviously something got totally mixed up as I tried to squash commits and revert old clang-stuff. I hope I've done everything correct now. At least it builds fine for me and I can use this method.

**Review by howie-f — APPROVED** [review]
nice 👍

**Comment by DaVukovic** [comment]
Missed that one totally. I'll have a look soon and will probably rebase it. If not, I'll close this one

**Pull request closed without merging**